### PR TITLE
TCD-1294: New SC client info endpoint

### PR DESCRIPTION
**Pull request by LoryApiFortress**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by LoryApiFortress** [comment]
@lysannep @waggledans please review. Thanks

File: docs/dev/api/connect.md [hunk 1]
```diff
@@ -287,6 +287,365 @@ curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
 
 ---
 
+### Get Tunnel Versions Information
```
Comment (lysannep) [inline docs/dev/api/connect.md:290]:
```suggestion
### Get Tunnel Versions
```

File: docs/dev/api/connect.md [hunk 2]
```diff
@@ -287,6 +287,365 @@ curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
...
+
+Returns tunnel version information.
+
+#### Parameters
+
+<table id="table-api">
+  <tbody>
+    <tr>
+     <td><code>client_version</code></td>
+    <td><p><small>| QUERY | OPTIONAL | STRING |</small></p><p>The Sauce Connect client version (For example, <code>4.7.1</code>).</p></td>
```
Comment (lysannep) [inline docs/dev/api/connect.md:303]:
```suggestion
    <td><p><small>| QUERY | OPTIONAL | STRING |</small></p><p>Returns download information for the specified Sauce Connect client version (For example, <code>4.7.1</code>).</p></td>
```

File: docs/dev/api/connect.md [hunk 3]
```diff
@@ -287,6 +287,365 @@ curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
...
+{
+   "download_url" : "https://saucelabs.com/downloads/sc-4.8.2-osx.zip",
+   "downloads" : {
+      "linux" : {
+         "download_url" : "https://saucelabs.com/downloads/sc-4.8.2-linux.tar.gz",
+         "sha1" : "<hash>"
+      },
+      "linux-arm64" : {
+         "download_url" : "https://saucelabs.com/downloads/sc-4.8.2-linux-arm64.tar.gz",
+         "sha1" : "fd782a658f4d28b9792ed<hash>af9df730a87ae797cba"
```
Comment (lysannep) [inline docs/dev/api/connect.md:597]:
```suggestion
         "sha1" : "<hash>"
```

File: docs/dev/api/connect.md [hunk 4]
```diff
@@ -287,6 +287,365 @@ curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
...
+
+<table id="table-api">
+  <tbody>
+    <tr>
+     <td><code>client_version</code></td>
+    <td><p><small>| QUERY | OPTIONAL | STRING |</small></p><p>The Sauce Connect client version (For example, <code>4.7.1</code>).</p></td>
+    </tr>
+    <tr>
+     <td><code>client_host</code></td>
+     <td><p><small>| QUERY | OPTIONAL | STRING |</small></p><p>The Sauce Connect host OS and CPU Architecture (For example, <code>darwin-amd64</code>). </p></td>
```
Comment (lysannep) [inline docs/dev/api/connect.md:307]:
```suggestion
     <td><p><small>| QUERY | OPTIONAL | STRING |</small></p><p>Returns download information for the specified Sauce Connect host OS and CPU Architecture (For example, <code>darwin-amd64</code>). </p></td>
```

**Review by lysannep — COMMENTED** [review]
@LoryApiFortress I added some suggestions and a couple questions.

File: docs/dev/api/connect.md [hunk 5]
```diff
@@ -287,6 +287,365 @@ curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
...
+   },
+   "info_url" : "https://docs.saucelabs.com/secure-connections/sauce-connect/installation",
+   "latest_version" : "4.8.2",
+   "warning" : [
+      "Client host platform is not specified, the download URL cannot be determined"
+   ]
+}
+```
+
+```jsx title="Sample Response to the request with an optional parameter all=true"
```
[the remark this answers is not in the document]
Comment (LoryApiFortress) [reply docs/dev/api/connect.md:405]: I copied/pasted the response from my call and the 4.6.x versions are returned. @waggledans is this intentional or is it a bug?

File: docs/dev/api/connect.md [hunk 6]
```diff
@@ -287,6 +287,365 @@ curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
...
+}
+
+```
+
+```jsx title="Sample Response to the request with an optional parameter client_version=4.7.1"
+{
+   "client_version" : "4.7.1",
+   "downloads" : {
+      "linux" : {
+         "download_url" : "https://saucelabs.com/downloads/sc-4.8.2-linux.tar.gz",
```
[the remark this answers is not in the document]
Comment (LoryApiFortress) [reply docs/dev/api/connect.md:620]: I copied/pasted the response from my call and this is what is returned. @waggledans is this intentional or is it a bug?

**Review by lysannep — APPROVED** [review]

File: docs/dev/api/connect.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (aliciakrebs) [reply docs/dev/api/connect.md:405]: @LoryApiFortress, Can you create another PR is needed once these questions are answered?

[the remark this answers is not in the document]
Comment (LoryApiFortress) [reply docs/dev/api/connect.md:405]: @aliciakrebs I asked those questions in the SC jira project, once I get the answers I'll open the PR if needed

**Pull request merged**